distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso

```diff
@@ -110,48 +120,632 @@ type Array
+    join self separator="" prefix="" suffix="" = Vector.join self separator prefix suffix
+
+    to_text : Text
+    to_text self = self.map .to_text . join ", " "[" "]"
+
+    ## Combines all the elements of a non-empty array using a binary operation.
+       If the array is empty, it returns `if_empty`.
+
+       Arguments:
+       - function: A binary operation that takes two items and combines them.
+       - if_empty: Value returned if the array is empty.
+
+       > Example
+         Compute the sum of all the elements in an array.
+
+             [0, 1, 2].to_array . reduce (+)
+    reduce : (Any -> Any -> Any) -> Any -> Any
+    reduce self function ~if_empty=(Error.throw Empty_Error) = Vector.reduce self function if_empty
+
+    ## Checks whether a predicate holds for at least one element of self array.
+
+       Arguments:
+       - predicate: A function that takes a list element and returns a boolean
+         value that says whether that value satisfies the conditions of the
+         function.
+
+       > Example
+         Checking if any element of the list is larger than 3.
+
+             [1, 2, 3, 4, 5].to_array.any (> 3)
+    any : (Any -> Boolean) -> Boolean
+    any self predicate = Vector.any self predicate
+
+    ## Checks whether a predicate holds for all elements in this array.
+
+       Arguments:
+       - predicate: A function that takes a list element and returns a boolean
+         value that says whether that value satisfies the conditions of the
+         function.
+
+       > Example
+         Check if all elements in the array are less than zero.
+
+             [-1, 1, 5, 8].to_array.all (< 0)
+    all : (Any -> Boolean) -> Boolean
+    all self predicate = self . any (predicate >> .not) . not
+
+    ## Checks whether this array contains a given value as an element.
+
+       Arguments:
+       - elem: The item to see if it exists in the array.
+
+       > Example
+         Checking if the array contains the number 72.
+
+             [1, 383, 72, 301].to_array.contains 72
+    contains : Any -> Boolean
+    contains self elem = self.any ix-> ix == elem
+
+    ## Performs a pair-wise operation passed in `function` on consecutive
+       elements of `self` and `that`.
+
+       Arguments:
+       - that: The array to zip with `self`.
+       - function: The function used to combine pairwise elements of `self` and
+         `that`.
+
+       The result of this function is a `Vector` of length being the shorter of
+       `self` and `that`, containing results of calling `function`.
+
+       > Example
+         Calculate the pairwise sum of two arrays.
+
+             [1, 2, 3].to_array.zip [4, 5, 6].to_array (+)
+
+       > Example
+         When the `function` is not provided, it defaults to creating a pair
+         of both elements.
+
+             [1, 2, 3].to_array.zip [4, 5, 6].to_array == [[1, 4].to_array, [2, 5].to_array, [3, 6].to_array].to_array
+    zip : Vector Any -> (Any -> Any -> Any) -> Vector Any
```

Comment:
```suggestion
    zip : Array Any -> (Any -> Any -> Any) -> Vector Any
```

Reply:
Again. The input type is either `Vector` or `Array`.

Reply:
Moreover that applies to the whole codebase. Everywhere `Vector` is used, `Array` can be used as well!

Rather than modifying all the type signatures, let's hope further consolidation renders these two types obsolete and allows them to be merged into one. Because `Array` is basically `Vector in Input` - e.g. `Vector` that can change due to changes made elsewhere and may need to be read (`Input`ed) again.